ooniapi/services/ooniprobe/src/ooniprobe/routers/v1/probe_services.py

```diff
@@ -0,0 +1,126 @@
+    # Maybe using the same exporter as the instrumentator?
+    try:
+        dec = decode_jwt(token, audience="probe_login", key=settings.jwt_encryption_key)
+        registration_time = dec["iat"]
+        log.info("probe login: successful")
+        # metrics.incr("probe_login_successful")
+    except jwt.exceptions.MissingRequiredClaimError:
+        log.info("probe login: invalid or missing claim")
+        # metrics.incr("probe_login_failed")
+        raise HTTPException(status_code=401, detail="Invalid credentials") 
+    except jwt.exceptions.InvalidSignatureError:
+        log.info("probe login: invalid signature")
+        # metrics.incr("probe_login_failed")
+        raise HTTPException(status_code=401, detail="Invalid credentials") 
+    except jwt.exceptions.DecodeError:
+        # Not a JWT token: treat it as a "legacy" login
+        # return jerror("Invalid or missing credentials", code=401)
+        log.info("probe login: legacy login successful")
+        # metrics.incr("probe_legacy_login_successful")
+        registration_time = None
+
+    exp = datetime.now(timezone.utc) + timedelta(days=7)
+    payload = {"registration_time": registration_time, "aud": "probe_token"}
+    token = create_jwt(payload, key=settings.jwt_encryption_key)
+    # expiration string used by the probe e.g. 2006-01-02T15:04:05Z
+    expire = exp.strftime("%Y-%m-%dT%H:%M:%SZ")
+    login_response = ProbeLoginResponse(token=token, expire = expire)
+    setnocacheresponse(response)
+
+    return login_response
+
+class ProbeRegister(BaseModel):
+    # None of this values is actually used, but I add them
+    # to keep it compliant with the old api
+    password : str
+    platform : str
+    probe_asn : str
+    probe_cc : str
+    software_name : str 
+    software_version : str
+    supported_tests : List[str]
+
+class ProbeRegisterResponse(BaseModel):
+    client_id: str 
+
+@router.post("/register", tags=["ooniprobe"], response_model=ProbeRegisterResponse)
+def probe_register_post(
+    probe_register : ProbeRegister, 
+    response : Response,
+    settings : Settings = Depends(get_settings),
+) -> ProbeRegisterResponse:
+    """Probe Services: Register
+
+    Probes send a random string called password and receive a client_id
+
+    The client_id/password tuple is saved by the probe and long-lived
+
+    Note that most of the request body arguments are not actually 
+    used but are kept here to use the same API as the old version
+    """
+    global log
+
+    # client_id is a JWT token with "issued at" claim and
+    # "audience" claim. The "issued at" claim is rounded up.
+    issued_at = int(time.time())
+    payload = {"iat": issued_at, "aud": "probe_login"}
+    client_id = create_jwt(payload, key=settings.jwt_encryption_key)
+    log.info("register successful")
+
+    register_response = ProbeRegisterResponse(client_id=client_id)
+    setnocacheresponse(response)
+
+    return register_response
+
+class ProbeUpdate(BaseModel):
```

Comment:
The ProbeUpdate request contains some useful metadata that we could use for the purpose of the anonymous credential anomaly detection to establish a baseline for comparing the probe_cc and probe_asn inconsistencies (see: https://github.com/ooni/orchestra/blob/master/registry/registry/handler/registry.go#L25).

Basically what I'm thinking of is that, if we assume that the update requests are not going to be tampered with as much as the measurement submit ones, we can use this to determine which inconsistencies are due to our geoip being off compared to the probe one, vs those which are due to the measurement metadata being inconsistent due to malice.

We can do parsing and logging of this as a follow up PR though.